### PR TITLE
Added REST API view to list courses in repository

### LINKED ISSUE #492: REST API for listing courses in a repository
Not absolutely necessary for #469, but it will be very hard to use without a course id

### PR DESCRIPTION
**Pull request by giocalitri**
Closes #492

### CODE REVIEW

File: rest/serializers.py [hunk 1]
```diff
@@ -61,6 +62,29 @@ class Meta:
         )
 
 
+class CourseSerialier(ModelSerializer):
+    """Serializer for Repository."""
```
Comment (noisecapella) [inline rest/serializers.py]: Should be `Serializer for Course`

File: rest/serializers.py [hunk 2]
```diff
@@ -61,6 +62,29 @@ class Meta:
...
+        slug_field="slug",
+        read_only=True
+    )
+
+    class Meta:
+        # pylint: disable=missing-docstring
+        model = Course
+        fields = (
+            'id',
+            'repository',
```
Comment (noisecapella) [inline rest/serializers.py]: Since this is in the URL as a slug we should probably remove it here

**Comment by giocalitri** [comment]
I forgot to edit the API docs

File: rest/tests/test_course.py [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+    def test_list_courses(self):
+        """
+        Tests the api to list courses in a repo
+        """
+        res = self.get_courses(repo_slug=self.repo.slug)
+        self.assertEqual(res['count'], 1)
+
+        # check content returned
+        course_1 = res['results'][0]
+        self.assertEqual(
```
Comment (noisecapella) [inline rest/tests/test_course.py]: Could you make this into a single assertEqual with two dictionaries instead?

File: rest/tests/test_course.py [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+            sorted(['id', 'repository', 'org', 'course_number', 'run'])
+        )
+        self.assertEqual(course_1['id'], self.course.id)
+        self.assertEqual(course_1['repository'], self.course.repository.slug)
+        self.assertEqual(course_1['org'], self.course.org)
+        self.assertEqual(course_1['course_number'], self.course.course_number)
+        self.assertEqual(course_1['run'], self.course.run)
+
+        # create another course
+        couse_params = {
```
Comment (noisecapella) [inline rest/tests/test_course.py]: `couse` should be `course`

File: rest/serializers.py [hunk 5]
```diff
@@ -61,6 +62,29 @@ class Meta:
         )
 
 
+class CourseSerialier(ModelSerializer):
```
Comment (noisecapella) [inline rest/serializers.py]: Should be `CourseSerializer`

**Comment by giocalitri** [comment]
@noisecapella worked on all comments

**Comment by noisecapella** [comment]
:+1:

**Comment by giocalitri** [comment]
Merging

**Pull request merged**